### PR TITLE
[SPARK-7792] [SQL] HiveContext registerTempTable not thread safe

### PR DESCRIPTION
**Pull request by navis**
Just replaced mutable.HashMap to ConcurrentHashMap

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala [hunk 1]
```diff
@@ -81,7 +84,7 @@ trait Catalog {
 }
 
 class SimpleCatalog(val conf: CatalystConf) extends Catalog {
-  val tables = new mutable.HashMap[String, LogicalPlan]()
+  val tables = new ConcurrentHashMap[String, LogicalPlan] asScala
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala]: Actually it's a bad idea to put a scala wrapper around java's ConcurrentHashMap. Do you mind just removing the wrapper, and change the call site to just use the functions defined in ConcurrentHashMap?

Also would be great to mark tables as a private val.

Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala]: For more background, see https://github.com/databricks/scala-style-guide#concurrency

Comment (navis) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala]: Oh, is it? Thanks. I got it.

**Comment by rxin** [comment]
Jenkins, test this please.

**Comment by rxin** [comment]
Jenkins, test this please.

**Comment by rxin** [comment]
Thanks. I'm going to merge this into master.

**Pull request closed without merging**